x-pack/plugins/lens/public/editor_frame_service/editor_frame/config_panel/add_layer.tsx

```diff
@@ -111,7 +120,26 @@ export function AddLayerButton({
               return {
                 toolTipContent,
                 disabled,
-                name: label,
+                name:
+                  type === layerTypes.ANNOTATIONS ? (
+                    <EuiFlexGroup gutterSize="xs">
+                      <EuiFlexItem grow={false}>
+                        <span className="lnsLayerAddButtonLabel">{label}</span>
+                      </EuiFlexItem>
+                      <EuiFlexItem>
+                        <EuiBadge color="hollow">
+                          <span className="lnsLayerAddButtonTechBadge">
+                            {i18n.translate('xpack.lens.configPanel.experimentalLabel', {
+                              defaultMessage: 'Technical preview',
+                            })}
+                          </span>
+                        </EuiBadge>
+                      </EuiFlexItem>
+                    </EuiFlexGroup>
```

Comment:
Suggesting the following changes to the `EuiFlexGroup` and its children. Changes include:

- Increasing the `gutterSize`.
- Moving `grow={false}` prop from the first to the second `EuiFlexItem`.
- Updating class names to use BEM naming.
- Moving `lnsLayerAddButton__techBadge` class to the parent `EuiBadge`.
- Utilizing the `isDisabled` prop for `EuiBadge` instead of custom styles.

```suggestion
                    <EuiFlexGroup gutterSize="m">
                      <EuiFlexItem>
                        <span className="lnsLayerAddButton__label">{label}</span>
                      </EuiFlexItem>

                      <EuiFlexItem grow={false}>
                        <EuiBadge
                          className="lnsLayerAddButton__techBadge"
                          color="hollow"
                          isDisabled={disabled}
                        >
                          {i18n.translate('xpack.lens.configPanel.experimentalLabel', {
                            defaultMessage: 'Technical preview',
                          })}
                        </EuiBadge>
                      </EuiFlexItem>
                    </EuiFlexGroup>
```